h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -270,7 +274,9 @@ private void parallelGridSearch(final Grid<MP> grid) {
     final ParallelModelBuilder parallelModelBuilder = new ParallelModelBuilder(modelFeeder);
 
     List<ModelBuilder> startModels = new ArrayList<>();
-    final List<MP> mps = initialModelParameters(_parallelism, iterator);
+
+    int numberOfInitialModels = iterator.max_models() == 0 ? _parallelism : Math.min(_parallelism, iterator.max_models());
```

Comment:
also had to take into account case where `_parallelism` is greater than `max_models()`